pkgs/development/python-modules/tweedledum/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, ninja
+, scikit-build
+  # Check Inputs
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "tweedledum";
+  version = "1.0.0";
+  format = "pyproject";
+
+  src = fetchFromGitHub{
+    owner = "boschmitt";
+    repo = "tweedledum";
+    rev = "v${version}";
+    hash = "sha256-59lJzdw9HLJ9ADxp/a3KW4v5aU/dYm27NSYoz9D49i4=";
+  };
+
+  nativeBuildInputs = [ cmake ninja scikit-build ];
+  dontUseCmakeConfigure = true;
+
+  pythonImportsCheck = [ "tweedledum" ];
+
+  # TODO: use pytest, but had issues with finding the correct directories
+  checkPhase = ''
+    python -m unittest discover -s ./python/test -t .
+  '';
```

Comment:
set pytestCheckFlags to ``python/test``.

Reply:
With the suggested change:
```
________________ ERROR collecting python/test/test_BitVector.py ________________
ImportError while importing test module '/build/source/python/test/test_BitVector.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/5470xw15wnn972ap0c4f7q642z4nvh6f-python3-3.8.8/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
python/test/test_BitVector.py:10: in <module>
    from tweedledum.bool_function_compiler import BitVec
python/tweedledum/__init__.py:5: in <module>
    from ._tweedledum import *
E   ModuleNotFoundError: No module named 'tweedledum._tweedledum'
______________ ERROR collecting python/test/test_BoolFunction.py _______________
ImportError while importing test module '/build/source/python/test/test_BoolFunction.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/5470xw15wnn972ap0c4f7q642z4nvh6f-python3-3.8.8/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
python/test/test_BoolFunction.py:7: in <module>
    from tweedledum.bool_function_compiler import BitVec, BoolFunction
python/tweedledum/__init__.py:5: in <module>
    from ._tweedledum import *
E   ModuleNotFoundError: No module named 'tweedledum._tweedledum'
_________________ ERROR collecting python/test/test_Circuit.py _________________
ImportError while importing test module '/build/source/python/test/test_Circuit.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/5470xw15wnn972ap0c4f7q642z4nvh6f-python3-3.8.8/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
python/test/test_Circuit.py:7: in <module>
    from tweedledum.ir import Circuit
python/tweedledum/__init__.py:5: in <module>
    from ._tweedledum import *
E   ModuleNotFoundError: No module named 'tweedledum._tweedledum'
=========================== short test summary info ============================
ERROR python/test/test_BitVector.py
ERROR python/test/test_BoolFunction.py
ERROR python/test/test_Circuit.py
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 3 errors in 0.37s ===============================
builder for '/nix/store/m90fn5fjd3wld2j1hqnky7j115xhn1cx-python3.8-tweedledum-1.0.0.drv' failed with exit code 2
```

Reply:
Seems like it the package won't work with pytest.